### DISCUSSION TITLE
Custom template and hot reload

### DISCUSSION THREAD

**Discussion by ArthurGoupil** [web-infra-dev/rsbuild#3819] [General]
Hello!

I'm trying to setup a SSR server where the html template is computed programmatically, and used in the dev server after.
The issue with not using the `html.template` field of rsbuild config is that the hot reload script is not injected into my template head.
Is there a way to do it manually?

Thanks!

**Comment by 9aoy** [comment]
The hot reload script is not injected into the HTML template, but into the entry script. 
Currently only client code supports hot reload, and pure SSR hot reload is still under design.

**Reply by ArthurGoupil** [reply]
Thanks for the reply!

**Reply by damassi** [reply]
Looking forward to SSR hot reloading 🙏
